### PR TITLE
[Feature] Added filter options for checking if the value is empty or not

### PR DESCRIPTION
**Pull request by MaksymShokin**
@sofiiakvasnevska

## Issue
https://github.com/Fliplet/fliplet-studio/issues/4800

## Description
Added possibility to filter by empty or not empty value. Value field will be hidden if empty or not empty field is selected.

## Screencast
https://streamable.com/usc02a

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/utils.js [hunk 1]
```diff
@@ -394,6 +394,14 @@ Fliplet.Registry.set('dynamicListUtils', (function () {
 
     return _.filter(records, function (record) {
       return _.every(filters, function (filter) {
+        if (filter.condition === 'empty') {
+          return _.isEmpty(record.data[filter.column]);
```
Comment (tonytlwu) [inline js/utils.js:398]: Numbers will be incorrectly reported as empty using `_.isEmpty()`.

Also, `NaN` is reported as a number using `_.isNumber()`.

This means:

```suggestion
          return _.isEmpty(record.data[filter.column]) && (_.isNaN(record.data[filter.column]) || !_.isNumber(record.data[filter.column]));
```

Also, use variables laid out in https://github.com/Fliplet/fliplet-widget-dynamic-lists/blob/feat/filter-empty/js/utils.js#L397-L400 which will help simplify the code.

Comment (MaksymShokin) [reply js/utils.js]: Fixed. I used isFinite lodash method

File: js/utils.js [hunk 2]
```diff
@@ -394,28 +394,33 @@ Fliplet.Registry.set('dynamicListUtils', (function () {
...
       return _.every(filters, function (filter) {
-        if (filter.condition === 'empty') {
-          return _.isEmpty(record.data[filter.column]);
-        }
-          
-        if (filter.condition === 'notempty') {
-          return !_.isEmpty(record.data[filter.column]); 
-        }
+        var condition = filter.condition;
+        var rowData = _.get(record, 'data.' + filter.column, null);
```
Comment (tonytlwu) [inline js/utils.js:398]: Using a string path ends up using `String.prototype.split()` which does not perform very well. This isn't usually a problem, but on an operation like this where the operation is executed over a potentially very large list, it's more efficient to pass an array as path.

```suggestion
        var rowData = _.get(record, ['data', filter.column], null);
```

File: js/utils.js (hunk 1 above)
Comment (tonytlwu) [reply js/utils.js]: Didn't know about `_.isFinite()`. Thanks for teaching me something!

File: js/utils.js (hunk 2 above)
Comment (MaksymShokin) [reply js/utils.js]: Fixed

**Comment by MaksymShokin** [comment]
@tonytlwu 
I tested all optioned you have mentioned. Now it should work correctly.

File: js/utils.js [hunk 3]
```diff
@@ -394,20 +394,33 @@ Fliplet.Registry.set('dynamicListUtils', (function () {
...
+
+        if (condition === 'none' || filter.column === 'none') {
           // Filter isn't configured correctly
           return true;
         }
 
-        var condition = filter.condition;
-        var rowData;
+        if (condition === 'empty') {
+          return _.isEmpty(rowData) && !_.isFinite(rowData) && rowData !== true && rowData !== false;
```
Comment (tonytlwu) [inline js/utils.js:406]:
```suggestion
          return _.isEmpty(rowData) && !_.isFinite(rowData) && typeof rowData !== 'boolean';
```

Comment (MaksymShokin) [reply js/utils.js]: Fixed

**Pull request merged**